EIPS/eip-7619.md

```diff
@@ -0,0 +1,251 @@
+    ) public returns (bool isValid) {
+        (bool success, bytes memory verifies) = address(0x0000000000000000000000000000000000000065)
+            .staticcall(
+                abi.encodeWithSignature(
+                    "verify(bytes,bytes,bytes)",
+                    signature,
+                    publicKey,
+                    message
+                )
+            );
+        require(success && verifies.length == 32, "Invalid signature");
+        return verifies[31] == 0;
+    }
+```
+
+### Gas costs and benchmarks
+
+Since data size is variable, each operation to execute the precompiled call will require:
+
+- 1465 base units of gas for each operation.
+- 6 units of gas for each for each word contained in the payload.
+  These values are based on data presented at the benchmark section.
+
+## Rationale
+
+Falcon-512 is a good candidate for a new signature implementation. First of all, key sizes are relatively small compared to ther signature algorithms that are also quantum resistant.
+
+The following table shows a summary of the typicall values for keys and the approximate amount of time taken per second to execute signature verifications as specified by the authors of Falcon algorithm:
+
+```
+variant                                     verify/s   pub size(bytes)  sig size(bytes)
+-----------------------------------------  ---------  ----------------  ---------------
+Falcon512                                    27933.0               897              666
+```
+
+Falcon uses shake256 under the hood, an implementation made in solidity showed it is not viable to implement this algorithm at that layer, on the other hand the implementation of falcon-512 by using precompiles turned to be the best approach since the gas cost (as show in the previous section) is relatively low.
+
+Implementing just the signature verification is sufficiently enough to open a wide range of use cases at the smart contract level. To give an example, this method can perfectly work with the Account Abstraction concept, where the authentication process would rely on Falcon and the account is a smart contract.
+
+## Backwards Compatibility
+
+There is very little risk of breaking backwards-compatibility with this EIP, the sole issue being if someone were to build a contract relying on the address at `0x65` being empty. The likelihood of this is low, and should specific instances arise, the address could be chosen to be any arbitrary value with negligible risk of collision.
+
+## Test Cases
+
+- A set of test vectors, with valid signatures, for benchmarking on new implementations is located in a separate [file](../assets/eip-7619/bench_vectors.md)
+
+- A set of test vectors, with invalid data, to verify some invalid signature cases is located in a separate [file](../assets/eip-7619/invalid_signature_test_vectors.md)
+
+- Additionally when method signature doesn't match with `de8f50a1` or just payload length is zero the precompile will throw an exception:
```

Comment:
This line is introducing a requirement so it should be in the Specification section.